src/explorers/SliceSampler.jl

```diff
@@ -23,24 +24,39 @@ end
 $SIGNATURES
 Slice sample one point.
 """
-function slice_sample!(h::SliceSampler, state::AbstractVector, log_potential, rng)
-    for c in 1:length(state) # update *every* coordinate
-        g_x0 = log_potential(state)
+
+function get_initial_logp(state::AbstractVector, log_potential)
+    return log_potential(state)
+end
+
+function get_initial_logp(state::DynamicPPL.TypedVarInfo, log_potential)
+    # for Turing, need to transform to get initial logp for caching
+    cached_lp = on_transformed_space(state, log_potential) do
+        return log_potential(state)
+    end
+    return cached_lp
+end
+
+function slice_sample!(h::SliceSampler, state::AbstractVector, log_potential, cached_lp, rng)
+    for c in 1:length(state) # update every coordinate
         pointer = Ref(state, c)
-        slice_sample_coord!(h, state, pointer, log_potential, g_x0, rng)
+        cached_lp = slice_sample_coord!(h, state, pointer, log_potential, cached_lp, rng)
```

Comment:
If I understand correctly, there might be an extra call to on_transformed involved. So on one end the cache would help, but on the other hand, the extra call to on_transformed might remove some of the savings. It would be interesting to do a quick and dirty benchmark to confirm the caching does pay off.

Reply:
Yes! I was sort of working under the assumption that `on_transformed_space` is not terribly expensive and that doing it once more is better than doing `log_potential` (d+2) times (which is how many fewer calls to `log_potential` the new code does, at a minimum) 

But we should def benchmark this on two examples: one where `log_potential` is cheap, and one where it's expensive

Reply:
I would be interested in seeing results from a benchmark because `on_transformed_space` might not be cheap compared to `log_potential` in all settings.

Reply:
as of 2f47002 , no more extra `on_transformed_space` call